### PR TITLE
Add all OIDC flags

### PR DESCRIPTION
**Pull request by pipo02mix**
Towards https://github.com/giantswarm/roadmap/issues/586

### CODE REVIEW

File: helm/cluster-openstack/values.yaml [hunk 1]
```diff
@@ -11,7 +11,11 @@ nodeCIDR: ""  # The OpenStack Subnet to be created.
 externalNetworkID: ""  # UUID of external network (optional).
 
 oidc:
-  enabled: false
+  issuer-url: ""
+  ca-file: ""
+  client-id: ""
+  username-claim: ""
+  groups-claim: ""
```
Comment (tfussell) [inline helm/cluster-openstack/values.yaml:18]:
```suggestion
  issuerUrl: ""
  caFile: ""
  clientId: ""
  usernameClaim: ""
  groupsClaim: ""
```

File: helm/cluster-openstack/templates/_kubeadm_control_plane_template.tpl [hunk 2]
```diff
@@ -16,11 +16,14 @@ spec:
...
-              oidc-username-claim: email
-              oidc-groups-claim: groups
+              {{- if .Values.oidc.issuer-url }}
+              oidc-issuer-url: {{ .Values.issuer-url }}
+              oidc-client-id: {{ .Values. client-id }}
+              oidc-username-claim: {{ .Values.username-claim }}
+              oidc-groups-claim: {{ .Values.groups-claim }}
+              {{- if .Values.oidc.ca-file }}
+              oidc-ca-file: {{ .Values.ca-file }}
+              {{- end }}
```
Comment (tfussell) [inline helm/cluster-openstack/templates/_kubeadm_control_plane_template.tpl:26]:
```suggestion
              {{- if .Values.oidc.issuerUrl }}
              oidc-issuer-url: {{ .Values.issuerUrl }}
              oidc-client-id: {{ .Values. clientId }}
              oidc-username-claim: {{ .Values.usernameClaim }}
              oidc-groups-claim: {{ .Values.groupsClaim }}
              {{- if .Values.oidc.caFile }}
              oidc-ca-file: {{ .Values.caFile }}
              {{- end }}
```

File: helm/cluster-openstack/templates/_kubeadm_control_plane_template.tpl [hunk 3]
```diff
@@ -16,11 +16,14 @@ spec:
...
             extraArgs:
               cloud-provider: external
-              {{- if .Values.oidc.enabled }}
-              oidc-issuer-url: https://dex.{{ include "resource.default.name" $ }}.{{ .Values.baseDomain }}
-              oidc-client-id: dex-k8s-authenticator
-              oidc-username-claim: email
-              oidc-groups-claim: groups
+              {{- if .Values.oidc.issuerUrl }}
+              oidc-issuer-url: {{ .Values.issuerUrl }}
+              oidc-client-id: {{ .Values. clientId }}
```
Comment (tfussell) [inline helm/cluster-openstack/templates/_kubeadm_control_plane_template.tpl:21]:
```suggestion
              oidc-client-id: {{ .Values.clientId }}
```

**Comment by pipo02mix** [comment]
@tfussell can we merge this?

**Review by tfussell — APPROVED** [review]

**Pull request merged**